keytree/src/lib.rs

```diff
@@ -78,6 +108,39 @@ impl Xpub {
             precompressed_pubkey: child_point.compress(),
         }
     }
+
+    /// Serializes this Xpub to a sequence of bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut buf = [0u8; 64];
+        buf[..32].copy_from_slice(self.precompressed_pubkey.as_bytes());
+        buf[32..].copy_from_slice(&self.dk);
+        buf
+    }
+
+    /// Decodes an Xpub from a 64-byte array, and fails if the provided array is not
+    /// exactly 64 bytes, or if the compressed point fails to decompress.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 64 {
+            return None;
+        }
+
+        let pieces = bytes.split_at(32);
+        let precompressed_pubkey = CompressedRistretto::from_slice(&pieces.0[..]);
+        let mut dk: [u8; 32] = Default::default();
+        dk.copy_from_slice(&pieces.1[..]);
+
+        match precompressed_pubkey.decompress() {
```

Comment:
i'd factor out decompressing from Xpub creation to make code less entangled and easier to read:

```rust
let point = match precompressed_pubkey.decompress() { Some(p)=>p, None => return None };
Some(Xpub {
    point, dk, precompressed_pubkey
})
```